src/order/rel_classes.lean

```diff
@@ -76,6 +76,11 @@ protected theorem is_asymm.is_irrefl [is_asymm α r] : is_irrefl α r :=
 protected theorem is_total.is_trichotomous (r) [is_total α r] : is_trichotomous α r :=
 ⟨λ a b, or.left_comm.1 (or.inr $ total_of r a b)⟩
 
+protected theorem is_irrefl.is_refl (r) [is_irrefl α r] : is_refl α (λ x y, ¬ r x y) :=
+⟨@irrefl α r _⟩
+protected theorem is_refl.is_irrefl (r) [is_refl α r] : is_irrefl α (λ x y, ¬ r x y) :=
+⟨λ a, not_not_intro (refl a)⟩
```

Comment:
If you move that to `order.boolean_algebra`, you can express it as.
```suggestion
protected theorem is_irrefl.compl (r) [is_irrefl α r] : is_refl α rᶜ := ⟨@irrefl α r _⟩
protected theorem is_refl.compl (r) [is_refl α r] : is_irrefl α rᶜ := ⟨λ a, not_not_intro (refl a)⟩
```

Reply:
Already discussed at https://github.com/leanprover-community/mathlib/pull/13411#discussion_r849303277

Reply:
I think I'm convinced this time around though. I'll move these in a sec